includes/class-attachments.php

```diff
@@ -0,0 +1,417 @@
+<?php
+/**
+ * Attachments processing file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub;
+
+use Activitypub\Collection\Posts;
+
+/**
+ * Attachments processor class.
+ */
+class Attachments {
+
+	/**
+	 * Initialize the class and set up filters.
+	 */
+	public static function init() {
+		\add_action( 'pre_get_posts', array( self::class, 'maybe_hide_from_media_library' ), 999 );
+		\add_action( 'before_delete_post', array( self::class, 'delete_attachments_with_post' ) );
+	}
+
+	/**
+	 * Delete attachments when an ap_post is deleted.
+	 *
+	 * @param int $post_id The post ID being deleted.
+	 */
+	public static function delete_attachments_with_post( $post_id ) {
+		if ( Posts::POST_TYPE !== \get_post_type( $post_id ) ) {
+			return;
+		}
+
+		foreach ( \get_attached_media( '', $post_id ) as $attachment ) {
+			\wp_delete_attachment( $attachment->ID, true );
+		}
+	}
+
+	/**
+	 * Hide ActivityPub attachments from Media Library queries.
+	 *
+	 * This works for both the list view and the media modal by checking
+	 * if we're querying attachments without explicitly requesting the
+	 * _activitypub_import meta key.
+	 *
+	 * @param \WP_Query $query The WordPress query object.
+	 */
+	public static function maybe_hide_from_media_library( $query ) {
+		// Only filter attachment queries.
+		if ( 'attachment' !== $query->get( 'post_type' ) ) {
+			return;
+		}
+
+		// Don't filter if we're querying attachments for an ap_post.
+		if ( $query->get( 'post_parent' ) ) {
+			$parent_post = \get_post( $query->get( 'post_parent' ) );
+			if ( $parent_post && Posts::POST_TYPE === $parent_post->post_type ) {
+				return;
+			}
+		}
+
+		// Check if the query is already explicitly looking for _activitypub_import.
+		$meta_query = $query->get( 'meta_query' ) ?: array(); // phpcs:ignore Universal.Operators.DisallowShortTernary
+
+		$has_activitypub_query = false;
+		foreach ( $meta_query as $clause ) {
+			if ( isset( $clause['key'] ) && '_activitypub_import' === $clause['key'] ) {
+				$has_activitypub_query = true;
+				break;
+			}
+		}
+
+		// If not explicitly querying for this meta, exclude ActivityPub imports.
+		if ( ! $has_activitypub_query ) {
+			$meta_query[] = array(
+				'key'     => '_activitypub_import',
+				'compare' => 'NOT EXISTS',
+			);
+			$query->set( 'meta_query', $meta_query );
+		}
+	}
+
+	/**
+	 * Process attachments from an ActivityPub object and attach them to a post.
+	 *
+	 * @param array $attachments Array of ActivityPub attachment objects.
+	 * @param int   $post_id     The post ID to attach files to.
+	 * @param int   $author_id   Optional. User ID to set as attachment author. Default 0.
+	 *
+	 * @return array Array of attachment IDs.
+	 */
+	public static function process( $attachments, $post_id, $author_id = 0 ) {
```

Comment:
`process` is a bit vague and does not really make clear what exactly the function is doing. Maybe `import_and_attach`?